### PR TITLE
SQ-76/Typography

### PR DESCRIPTION
**Pull request by tiwiz**
Add Calligraphy and fonts to the app

### CODE REVIEW

File: app/src/main/java/net/squanchy/fonts/TypefaceManager.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import net.squanchy.R;
+
+import uk.co.chrisjenx.calligraphy.CalligraphyConfig;
+import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper;
+
+public class TypefaceManager {
+
+    public static void init() {
+
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/fonts/TypefaceManager.java]: Why this empty line?

File: app/src/main/java/net/squanchy/fonts/TypefaceStyleableActivity.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package net.squanchy.fonts;
+
+import android.content.Context;
+import android.support.v7.app.AppCompatActivity;
+
+public class TypefaceStyleableActivity extends AppCompatActivity{
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/fonts/TypefaceStyleableActivity.java]: `cmd-alt-l`

File: app/src/main/res/values/fonts.xml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+  <string name="default_typeface_regular">fonts/Quicksand-Regular.ttf</string>
+  <string name="default_typeface_medium">fonts/Quicksand-Medium.ttf</string>
+  <string name="default_typeface_light">fonts/Quicksand-Light.ttf</string>
+  <string name="default_typeface_bold">fonts/Quicksand-Bold.ttf</string>
+  <string name="title_typeface">fonts/leaguespartan-bold.ttf</string>
+</resources>
```
Comment (rock3r) [inline app/src/main/res/values/fonts.xml]: Missing empty line

File: app/src/main/java/net/squanchy/fonts/TypefaceManager.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        CalligraphyConfig.initDefault(createConfigurationBuilder());
+    }
+
+    private static CalligraphyConfig createConfigurationBuilder() {
```
Comment (danybony) [inline app/src/main/java/net/squanchy/fonts/TypefaceManager.java]: This is creating the config, not the builder 😉

**Review by danybony — COMMENTED** [review]
Just a renamed method, and it LGTM

File: app/src/main/res/values/fonts.xml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+  <string name="default_typeface_regular">fonts/Quicksand-Regular.ttf</string>
+  <string name="default_typeface_medium">fonts/Quicksand-Medium.ttf</string>
+  <string name="default_typeface_light">fonts/Quicksand-Light.ttf</string>
+  <string name="default_typeface_bold">fonts/Quicksand-Bold.ttf</string>
+  <string name="title_typeface">fonts/leaguespartan-bold.ttf</string>
```
Comment (rock3r) [inline app/src/main/res/values/fonts.xml]: This was supposed to be called `leaguespartan-bold.otf`, I think you got the wrong version of the font

File: app/src/main/res/values/fonts.xml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+  <string name="default_typeface_regular">fonts/Quicksand-Regular.ttf</string>
```
Comment (rock3r) [inline app/src/main/res/values/fonts.xml:4]: Why not using attributes? This is not really what I had outlined... I meant:

```xml
In attrs.xml:
<attr name="titleTypeface" format="string" />
<attr name="defaultRegularTypeface" format="string" />
<attr name="defaultMediumTypeface" format="string" />
<attr name="defaultBoldTypeface" format="string" />
<attr name="defaultItalicTypeface" format="string" />

In themes.xml:
<style name="Theme.Squanchy">
  <item name="titleTypeface">fonts/leaguespartan-bold.otf</item>
  <item name="defaultRegularTypeface">fonts/Quicksand-Regular.ttf</item>
  <item name="defaultMediumTypeface">fonts/Quicksand-Medium.ttf</item>
  <item name="defaultBoldTypeface">fonts/Quicksand-Bold.ttf</item>
  <item name="defaultItalicTypeface">fonts/Quicksand-Italic.ttf</item>
</style>

In styles.xml:
<style name="Whatever.Title">
  <item name="fontPath">?attr:titleTypeface</item>
</style>
```

Would that not work?

File: dependencies.gradle [hunk 7]
```diff
@@ -28,6 +28,7 @@ ext {
                     timber           : 'com.jakewharton.timber:timber:4.3.1',
                     tweetUi          : 'com.twitter.sdk.android:tweet-ui:2.2.0',
                     viewPagerAdapter : 'com.novoda:view-pager-adapter:0.9.1',
+                    calligraphy      : 'uk.co.chrisjenx:calligraphy:2.2.0'
```
Comment (rock3r) [inline dependencies.gradle]: What about keeping alphabetic order?

**Review by rock3r — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/fonts.xml [hunk 8]
```diff
@@ -5,5 +5,6 @@
   <string name="default_typeface_medium">fonts/Quicksand-Medium.ttf</string>
   <string name="default_typeface_light">fonts/Quicksand-Light.ttf</string>
   <string name="default_typeface_bold">fonts/Quicksand-Bold.ttf</string>
-  <string name="title_typeface">fonts/leaguespartan-bold.ttf</string>
+  <string name="title_typeface">fonts/LeagueSpartan-Bold.otf</string>
+
 </resources>
```
Comment (rock3r) [inline app/src/main/res/values/fonts.xml]: Missing empty line... at EOF I meant xD

**Review by rock3r — APPROVED** [review]
LGTM but check you have the "ensure line feed at end of file" setting on!

**Pull request merged**